example/README.md

```diff
@@ -0,0 +1,342 @@
+    ```
+    <a id="running-service"></a>
+5. Start client with service: `npm run start`
+    ```bash
+    ...
+    > example@1.0.0 start
+    > node src/echo.js
+
+    📗 created a fluence peer 12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN with relay 12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi
+    Copy this code:
+    fluence run -f 'echoJS("12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN", "12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi", "echo", "msg")'
+    ```
+
+6. Open a new terminal and copy the last line and execute to check echo service:
+
+    ```bash
+    $ fluence run -f 'echoJS("12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN", "12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi", "echo", "msg")'
+    Running:
+      function: echoJS("12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN", "12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi", "echo", "msg")
+      relay: /dns4/kras-09.fluence.dev/tcp/19001/wss/p2p/12D3KooWD7CvsYcpF9HE9CCV9aY3SJ317tkXVykjtZnht2EbzDPm
+    ... done
+
+    Result:
+
+    "12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN: msg"
+    ```
+
+
+## Register both services in Registry
+
+If we, as service providers, want to be all echo services discoverable without specifying exact peer and service ids, we should use Registry to **advertise** our services.
+
+Firstly, we need to create a **resource.** A resource represents a group of service providers and has a corresponding resource id for its discovery. Secondly, we need to register service providers on this resource id. So echo services can be discovered and resolved by resource id only.
+
+1. The following [code](src/aqua/main.aqua) registers resource with label `echo`:
+
+    ```rust
+    func registerResource() -> ?string:
+        resource_id, error <- createResource("echo")
+        <- resource_id
+    ```
+
+2. Let’s register resource:
+    <a id="resource-id"></a>
+    ```bash
+    fluence run -f 'registerResource()'
+    ```
+
+    output:
+
+    ```bash
+    $ fluence run -f 'registerResource()'
+    Running:
+      function: registerResource()
+      relay: /dns4/kras-01.fluence.dev/tcp/19001/wss/p2p/12D3KooWKnEqMfYo9zvfHmqTLpLdiHXPe4SVqUWcWHDJdFGrSmcA
+    ... done
+
+    Result:
+
+    [
+      "echo12D3KooWRgEgxP4qAyUR5jerwKE1rwTLZoAhJ3YwAhpeSMC5pi77"
+    ]
+    ```
+
+    So the resource id is `echo12D3KooWRgEgxP4qAyUR5jerwKE1rwTLZoAhJ3YwAhpeSMC5pi77`. By this resource id we can access any registered provider of this resource. There could be more registered echo service on different peers that we can smoothly use.
+
+3. This [code](src/aqua/main.aqua) registers deployed service by given `resource_id`:
+
+    ```rust
+    func registerService(resource_id: string) -> bool:
+        services <- App.services()
+        echoService = services.echoService.default[0]
+        success, error <- registerNodeProvider(echoService.peerId, resource_id, "", ?[echoService.serviceId])
+        <- success
+    ```
+
```

Comment:
A short Aqua code explainer might be helpful here

Reply:
What every line of the code does. So, it's clear what's happening when you call the function